### ISSUE TITLE
[Minor suggestion for future video content] A video in regards to "workflow use" / computer setup, from the point of view of glimmer, but also aside from glimmer

### ISSUE THREAD

**Issue by rubyFeedback** [AndyObtiva/glimmer#33]
Hey there Andy,

On your latest blog entry you feature tetris (again). :D

And you also encouraged remarks e. g. "If you have any questions".

You are now at video #33.

I think most videos feature individual aspects of glimmer, e. g. 
widget use, different applications, code, style, showcasing,
demo - these things.

But I think it would be nice if you could make a "special"
video one day, perhaps at entry #40 or something that is
a bit different.

Ok so what do I mean with "different"?

Well, I actually mean this in the context of how YOU use the
computer. I mean both in regards to generic/general use
case, but also specifically for glimmer. Let me explain what
I mean.

I assume you use OSX primarily (I think); I am more used to
a linux workflow, so I literally use the commandline a lot.

Consider that you want to create a new application. I assume
you perhaps start with SWT (or libui, either way, this is just
as an example).

So what I think would be interesting to see would be LITERALLY
the steps you go through from CREATING a NEW application.

I understand that sometimes thinking takes place :D so I am not
saying EVERYTHING should be included in the video. But I think
it would be interesting to see how you approach this, via code
as-is. A bit like a "live" coding session - see the author of
Natalie who sometimes features videos similar to that.

For instance, I found this video accidentally some days ago:

https://www.youtube.com/watch?v=m-FEuur8xLo&list=PLWUx_XkUoGTq-nkbhnk6PN4m109ISo5BX&index=74

What to me was interesting was his workflow, e. g. vim + C++.

I think something like this may be interesting. I am not saying you should
copy that 1:1; the video on the head is (for me at the least) quite irrelevant,
but the code + workflow + commentary is really really neat.

I am also not suggesting you change your video style at all! I am more
saying "something a bit different".

So, what would we then, as viewer, see, or could learn from?

Well - the learning part is nice, so I think it would be nice to see how YOU
approach designing a new widget. Ideally something new, and something
that is not too simple and not too difficult. Just like literally something you
could create in 1-3 hours time (from start), and that would be something
useful. Like tetris. Or a game. Or some note-taking application or something.
It can be anything really. The important part is to show the usage pattern
and work flow.

IF possible it would be nice to see this written in SWT aka "glimmer in 
glimmer" but this is not so important compared to seeing the workflow.

The length, well - could be 20 minutes to 1 hours or so. I think more than
that is hard to follow. I don't watch all videos usually but select a subset
and scan through a bit when it is interesting.

Anyway. This is just a suggestion.You are probably mega-busy with 
different things, so perhaps this is for the next year. Or perhaps 
video #50 or something.

TL;DR: Please consider making a workflow video, related to glimmer
but also the general approach.

(PS: This may also be interesting for rails folks, and those who may
want to learn about rails, indirectly, in the sense that you have a 
rails background as well, whereas not every viewer has a rails
background. See kojix2 and the <= => data binding situation or
whatever it was. That was also new to me by the way. I know only
very little rails; I only built a few hello routes or show-the-time 
routes. But I wrote quite a bit of sinatra code so I am somewhat
familiar with sinatra.)

**Comment by rubycoder** [comment]
Great suggestion!!! I would also like to see a video like this, especially
if it uses SWT.

Wayne

**Comment by AndyObtiva** [comment]
@rubyFeedback Actually, this is a very well-timed issue. I was about to create a video similar to the idea you suggested in Glimmer DSL for SWT Video Tutorial 35 (only 2 videos away), which would begin by building a desktop application from scratch and progressing step by step until there is a fully working application. So, I really appreciate your suggestions (and the added confirmation of interest by @rubycoder ). I will keep them in mind when creating that video. 

I'm going to keep this issue open till I've addressed it in Glimmer DSL for SWT Video Tutorial 35. 

Cheers!

**Comment by AndyObtiva** [comment]
By the way, as a side-note, I have also taken note of your other issues ( #18 and #20 ) about creating a video that covers multiple Glimmer gems, comparing JRuby, CRuby, and Opal. So, there is no need to open further issues about that, but that video might not happen for a while. It won't be part of the Glimmer DSL for SWT Video Tutorial series.

**Comment by AndyObtiva** [comment]
I was about to build an API-based app from scratch and record a Glimmer DSL for SWT video about it today, but then I found out I had to register for that free API access and wait for approval.

So, I just did a Glimmer DSL for SWT tutorial (35) about scaffolding and packaging in the meantime.

**Comment by AndyObtiva** [comment]
Otherwise, I was on vacation 2 weekends ago, and I was working last weekend for a big software release for my employer. That's why I didn't post any tutorials during that period. In any case, the Glimmer DSL for SWT tutorials are pretty much done anyways.

**Comment by mahfuzullah** [comment]
@AndyObtiva very nice project my dude, 


i was building a little internal app using SWT dsl, i love how easy it is, but documentation could use some work,

i got stuck trying to set initial window size, (@rubyFeedback , you are welcome to tell me btw, still i found no documentation on this)

maybe we should open discord as user forum for glimmer
